tiledb/sm/filesystem/vfs_file_handle.cc

```diff
@@ -70,10 +69,9 @@ Status VFSFileHandle::close() {
     RETURN_NOT_OK(vfs_->close_file(uri_));
 
     // Create an empty file if the file does not exist
-    bool exists;
-    RETURN_NOT_OK(vfs_->is_file(uri_, &exists));
-    if (!exists)
-      RETURN_NOT_OK(vfs_->touch(uri_));
+    if (!vfs_->is_file(uri_)) {
+      vfs_->touch(uri_);
+    }
```

Comment:
```suggestion
    vfs_->touch(uri_);
```

`VFS::touch` itself makes the check for the file's existence. For the clouds where we do conditional requests, there might be a concern that we are always going to do a more expensive write operation, but I believe the reduced roundtrips are worth it, and this code path runs only in the user-facing VFS C API.